tasks/node_maintainance/node_health_monitor.py

```diff
@@ -0,0 +1,692 @@
+import sys
+import os
+
+# Adding project path into the sys paths for scanning all the modules
+script_dir = os.path.dirname(os.path.realpath(__file__))
+project_paths = [
+    os.path.join(script_dir, "..", ".."),
+    os.path.join(script_dir, "..", "..", "util", "ssh_util")
+]
+for project_path in project_paths:
+    if project_path not in sys.path:
+        sys.path.append(project_path)
+
+from helper.sdk_helper.testdb_helper.host_pool_helper import HostSDKHelper
+from helper.sdk_helper.testdb_helper.server_pool_helper import ServerPoolSDKHelper
+from util.ssh_util.node_infra_helper.remote_connection_factory import RemoteConnectionObjectFactory
+from constants.doc_templates import NODE_TEMPLATE
+import logging.config
+import paramiko, paramiko.ssh_exception
+import socket
+import concurrent
+import copy
+import datetime
+import json
+import argparse
+
+logger = logging.getLogger("tasks")
+
+# TODO tasks
+'''
+1. Checking state and move from booked to available if its in booked state for more than 48hrs
+2. Checking status of ntp
+3. Checking status of directories and permissions on the nodes
+4. Checking status of reserved nodes
+'''
+
+def check_connectivity_node(doc):
+    result = {}
+    ipaddr = doc["ipaddr"]
+
+    try:
+        server_pool_helper = ServerPoolSDKHelper()
+        logger.info(f"Connection to Server Pool successful")
+    except Exception as e:
+        result["result"] = False
+        result["reason"] = f"Cannot connect to Server Pool using SDK : {e}"
+        logger.error(result["reason"])
+        return result
+
+    if "tags" not in doc:
+        doc["tags"] = {}
+
+    try:
+        RemoteConnectionObjectFactory.fetch_helper(ipaddr,"root","couchbase")
+        doc["tags"]["connection_check"] = True
+        # doc["state"] = "available" if doc["state"] == "unreachable" else doc["state"]
+    except Exception as e:
+        doc["tags"]["connection_check"] = False
+        # doc["state"] = "unreachable"
+
+    try:
+        res = server_pool_helper.add_node_to_server_pool(doc)
```

Comment:
Change the function name since this is a bit ambiguous

Reply:
Addressed in latest commit